File: Regression/NRG/Enrollments/enroll_electric.py
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.select import Select
import time
import xlrd
import os

if os.environ.get('USE_PHANTOM'):
    print("using PhantomJS")

    driver = webdriver.PhantomJS()
    driver.implicitly_wait(5)
    driver.set_window_size(1120, 550)
else:
    print("using Firefox")

workbook = xlrd.open_workbook("./NRG_Data.xlsx")
worksheet = workbook.sheet_by_name('electric')

for current_row in range(1,worksheet.nrows):
    url = worksheet.row(current_row)[2].value
    first_name = worksheet.row(current_row)[3].value
    last_name = worksheet.row(current_row)[4].value
    email_addr = worksheet.row(current_row)[7].value
    confirm_email_addr = worksheet.row(current_row)[8].value
    Service_Address1 = worksheet.row(current_row)[9].value
    city = worksheet.row(current_row)[10].value
    zipcode = worksheet.row(current_row)[11].value
    Service_phone_number = worksheet.row(current_row)[12].value
    accountNo = worksheet.row(current_row)[13].value

    driver = webdriver.Firefox()    
    driver.get(url)
    time.sleep(2)

## Personal Information
    elem = driver.find_element_by_id("id_first_name").send_keys(first_name)
    elem = driver.find_element_by_id("id_last_name").send_keys(last_name)
    elem = driver.find_element_by_id("id_email").send_keys(email_addr)
    elem = driver.find_element_by_id("id_ver_email").send_keys(confirm_email_addr)
    elem = driver.find_element_by_id("id_phone").send_keys(Service_phone_number)
    elem = driver.find_element_by_id("id_service_address_1").send_keys(Service_Address1)
    elem = driver.find_element_by_id("id_service_address_city").send_keys(city)
    elem = driver.find_element_by_id("id_service_address_zip").send_keys(zipcode)
    elem = driver.find_element_by_id("id_electric-uan").send_keys(accountNo)
    try:
        elem = driver.find_element_by_id("id_electric-billing_uan").send_keys(accountNo)
    except:
        pass    
    time.sleep(2)
    elem = driver.find_element_by_id("continue-submit").click()
    time.sleep(4)

## Verification
    elem = driver.find_element_by_class_name("tos-section").send_keys(Keys.CONTROL, Keys.ARROW_DOWN);
    elem = driver.find_element_by_class_name("tos-section").send_keys(Keys.CONTROL, Keys.ARROW_DOWN);
    elem = driver.find_element_by_class_name("tos-section").send_keys(Keys.CONTROL, Keys.ARROW_DOWN);
    elem = driver.find_element_by_class_name("tos-section").send_keys(Keys.CONTROL, Keys.ARROW_DOWN);
    elem = driver.find_element_by_class_name("tos-section").send_keys(Keys.CONTROL, Keys.ARROW_DOWN);        
    elem = driver.find_element_by_id("id_order_authorization").click()

    if driver.find_element_by_id("id_affiliate_consent").is_displayed():
        driver.find_element_by_id("id_affiliate_consent").click()

    elem = driver.find_element_by_id("agree").click()

## Grab Conformation Code
    time.sleep(2)
    elem = driver.find_element_by_id("confirmation")
    confirmation = elem.text
    print("Passed - NRG_regression WEB Electric, Confirmation =  " +confirmation + ' for ' +last_name)

## Close Session:
    driver.close()

File: Regression/NRG/Enrollments/enroll_electric.rb
require 'rubygems'
require 'watir-webdriver'
require "win32ole"
require 'csv'

rows = 2	#2
while    
rows <= 30		#26

excel= WIN32OLE::new("excel.Application")
workbook=excel.Workbooks.Open("C:\\Scripts\\NRG\\Enrollments\\NRG_Data.xlsx")
worksheet = workbook.worksheets(1)

sku=worksheet.cells(rows,"B").value
url=worksheet.cells(rows,"C").value
first_name=worksheet.cells(rows,"D").value.to_s
last_name=worksheet.cells(rows,"E").value
email_addr=worksheet.cells(rows,"H").value
confirm_email_addr=worksheet.cells(rows,"I").value
Service_Address1=worksheet.cells(rows,"J").value
city=worksheet.cells(rows,"K").value.to_s
zip=worksheet.cells(rows,"L").value.to_s
Service_phone_number=worksheet.cells(rows,"M").value.to_s
accountNo=worksheet.cells(rows,"N").value.to_s
price=worksheet.cells(rows,"O").value
con1=worksheet.cells(rows,"P").value
con2=worksheet.cells(rows,"Q").value
con3=worksheet.cells(rows,"R").value

browser = Watir::Browser.new :firefox
string_from_excel = "goto"
browser.send string_from_excel, url
sleep 2

browser.text_field(:id => 'id_first_name').set first_name
browser.text_field(:id => 'id_last_name').set last_name
browser.text_field(:id => 'id_email').set email_addr
browser.text_field(:id => 'id_ver_email').set confirm_email_addr
browser.text_field(:id => 'id_phone').set("2152223333")
browser.text_field(:id => 'id_service_address_1').set Service_Address1
browser.text_field(:id => 'id_service_address_city').set city
browser.text_field(:id => 'id_service_address_zip').set zip
browser.checkbox(:id =>  'id_billing_same').clear
browser.checkbox(:id => 'id_billing_same').set
browser.text_field(:id => 'id_electric-uan').set accountNo
if
browser.text_field(:id => 'id_electric-billing_uan').exists?
browser.text_field(:id => 'id_electric-billing_uan').set accountNo
end
sleep 2
browser.button(:id => 'continue-submit').click
sleep 4

wsh = WIN32OLE.new('Wscript.Shell')
wsh.AppActivate('NRG Home Power')
browser.div(:class => "tos-section").send_keys [:control, :end]

browser.checkbox(:id, "id_order_authorization").set

if
browser.checkbox(:id, "id_affiliate_consent").exists?
browser.checkbox(:id, "id_affiliate_consent").set
end

browser.button(:value,"Submit").click

CSV.open('C:\\Scripts\\NRG\\Enrollments\\enroll_Conf_data.csv', 'ab') do |csv|
csv << [browser.div(:id => 'confirmation').text, sku=worksheet.cells(rows,"B").value, price=worksheet.cells(rows,"O").value,
con1=worksheet.cells(rows,"P").value, con2=worksheet.cells(rows,"Q").value, con3=worksheet.cells(rows,"R").value]
end

sleep 1
browser.close 
rows=rows+1

$end
workbook.Save
workbook.Close

end
$end
sleep 1